### PR TITLE
Add Language Analyze Documents APIs

### PR DESCRIPTION
**Pull request by aurghob**
# Data Plane API Specification Update Pull Request

Language today has analyze-text and analyze-conversations APIs to do NLP related tasks on the user inputs. We're introducing a new set of APIs that do similar tasks but on different type of inputs, native documents (E.g. .pdf, .docx, ..etc).

The API will perform the same tasks on the input, but a subset of them will reconstruct the document with updated content (E.g. PII redaction).

This API will use BYOS for accessing and returning the service results.

We already had a review of this change via https://github.com/Azure/azure-rest-api-specs-pr/pull/15838. 
This change is to make sure that the swagger is merged in the main repo as the review was for public preview.

## API Info: The Basics

Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue: https://github.com/Azure/azure-rest-api-specs/issues/26173

Is this review for (select one):

- [x] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document: https://microsofteur-my.sharepoint.com/:w:/g/personal/moaba_microsoft_com/Ee7_K5PvTg9Flg5mkgoAHRIBUuFwDL0sgxSduBg3Kovi4A?e=Y5pOcH
* Previous API Spec Doc: N/A
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories) 
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/cognitiveservices/data-plane/Language/preview/2023-11-15-preview/analyzedocuments.json [hunk 1]
```diff
@@ -0,0 +1,2237 @@
...
+    },
+    "PiiCategoriesExclude": {
+      "description": "(Optional) describes the PII categories to return",
+      "items": {
+        "type": "string",
+        "x-ms-enum": {
+          "name": "PiiCategoriesExclude",
+          "modelAsString": true
+        },
+        "enum": [
```
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2023-11-15-preview/analyzedocuments.json:1726]: Why does `piiCategories` use the `PiiCategory` enum while `PiiCategoriesExclude` does not? Is it a subset?

File: specification/cognitiveservices/data-plane/Language/preview/2023-11-15-preview/analyzedocuments.json [hunk 2]
```diff
@@ -0,0 +1,2237 @@
...
+      ]
+    },
+    "PiiDomain": {
+      "type": "string",
+      "description": "The PII domain used for PII Entity Recognition.",
+      "default": "none",
+      "enum": [
+        "phi",
+        "none"
+      ],
```
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2023-11-15-preview/analyzedocuments.json:1693]: You should be consistent in your casing for enums. You use both PascalCase and camelCase in this document.

**Review by heaths — APPROVED** [review]
A few comments, but assuming the service has already shipped this way, nothing blocking. I'd like those comments resolve first, however.

File: specification/cognitiveservices/data-plane/Language/preview/2023-11-15-preview/analyzedocuments.json (hunk 1 above)
Comment (aurghob) [reply specification/cognitiveservices/data-plane/Language/preview/2023-11-15-preview/analyzedocuments.json:1726]: PIIcategoriesExclude is a subset of PIIcategories. Its more cleanly expressed via the tsp but the swagger generated will be similar to what we have here.

File: specification/cognitiveservices/data-plane/Language/preview/2023-11-15-preview/analyzedocuments.json (hunk 2 above)
Comment (aurghob) [reply specification/cognitiveservices/data-plane/Language/preview/2023-11-15-preview/analyzedocuments.json:1693]: Currently keeping consistent with naming in analyzetext (naming is similar for the same enums). We will have a client.tsp for casing corrections that will make it more convenient for SDK release.

**Pull request merged**